src/com/amazon/ionschema/IonSchemaSystemBuilder.kt

```diff
@@ -100,23 +101,73 @@ class IonSchemaSystemBuilder private constructor() {
     }
 
     /**
-     * Allows forward-compatibility with the fixed, spec-compliant behavior for handling
-     * schema imports that is introduced in `ion-schema-kotlin-2.0.0`.
+     * Allows forward-compatibility with the fixed, spec-compliant behavior for
+     * handling schema imports that is introduced in `ion-schema-kotlin-2.0.0`.
+     * If not set, the default value is `true`.
+     *
+     * When set to `true`, the `IonSchemaSystem` will generate warnings about
+     * usages of transitive imports, which can be consumed by configuring the
+     * `IonSchemaSystem` with a callback function using [withWarningMessageCallback].
+     *
+     * Before setting this option to `false`, you should verify that your schemas
+     * will continue to work as intended, and you may need to update schemas that
+     * rely on the current, incorrect import resolution logic.
      *
      * @since 1.2
+     * @see [WarningType.INVALID_TRANSITIVE_IMPORT]
      */
     fun allowTransitiveImports(boolean: Boolean): IonSchemaSystemBuilder {
         params[IonSchemaSystemImpl.Param.ALLOW_TRANSITIVE_IMPORTS] = boolean
         return this
     }
 
     /**
-     * Provides a callback for the IonSchemaSystem to use for writing log messages.
+     * Provides a callback for the IonSchemaSystem to send a warning message about
+     * things that are not fatal (ie. will not result in an exception being thrown).
+     * Content of the messages may include information about possible errors in
+     * schemas, and usage of features that may be deprecated in future versions of
+     * the Ion Schema Language or the `ion-schema-kotlin` library.
+     *
+     * Clients can use these warnings as they please. Some possible uses are:
+     * - Log the warning messages
+     * - Surface the warnings to an end user who is authoring schemas
+     * - Enforce a "strict mode" by throwing an exception for warnings (ie. like
+     *   `javac -Werror`, but for Ion Schemas)
+     *
+     * NOTE—This method is equivalent to [withWarningMessageCallback], except that
+     * this method accepts a [Consumer]. This method is intended to be used for
+     * calling from Java code because [withWarningMessageCallback] requires a
+     * function that returns [Unit], which may be awkward for Java callers.
```

Comment:
Is using Consumer awkward for Kotlin callers?

Reply:
Using Consumer is not awkward for Kotlin—just more verbose than using a regular kotlin function. Both methods are usable from either language, they're just each un-idiomatic in one language, but clean and idiomatic in the other.

```java
// Java
IonSchemaSystemBuilder.standard()
        .withWarningMessageConsumer(System.out::println) // <-- idiomatic java
        .withWarningMessageCallback(it -> { System.out.println(it); return Unit.INSTANCE; })
```
```kotlin
// Kotlin
IonSchemaSystemBuilder.standard()
        .withWarningMessageConsumer(Consumer { println(it) })
        .withWarningMessageCallback { println(it) }  // <-- idiomatic Kotlin
```
I poked around a little more to see if there was any nice solution, but Java's `void` is not part of Kotlin's `Any`, so while I could create a functional interface that extends `(String) -> Unit` that doesn't solve the problem. This seems to be a rough edge that is specific to `Consumer`, but would not occur for a java `Supplier<T>` or java `Function<T, R>`. [Here](https://discuss.kotlinlang.org/t/java-interop-unit-closures-required-to-return-kotlin-unit/1842) is an interesting discussion of the problem, if you have time to start going down a rabbit hole.

That being said, it seems like it would be better for me to overload the function rather than have two different names as I do now. I'm going to change that.